data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/PluginArgumentsContext.java

```diff
@@ -56,15 +64,20 @@ private PluginArgumentsContext(final Builder builder) {
         if(typedArgumentsSuppliers.containsKey(parameterType)) {
             return typedArgumentsSuppliers.get(parameterType);
         }
-
-        throw new InvalidPluginDefinitionException("Unable to create an argument for required plugin parameter type: " + parameterType);
+        else if (applicationContext != null) {
+            return () -> applicationContext.getBean(parameterType);
```

Comment:
I'm somewhat concerned about providing *any* bean from the application context. Plugins could begin to rely on beans which we do not expect to provide.

I'd be up for either of the following changes:
1. Decorate the `BeanFactory` with a `BeanFactory` which only allows certain types. See my comment above.
2. Create two application contexts in Data Prepper. Provide the correct one into here. This would be along the lines of using application context inheritance to limit what we provide to plugins.